examples/md-flexible/src/Simulation.cpp

```diff
@@ -372,6 +465,106 @@ void Simulation::calculateGlobalForces(const std::array<double, 3> &globalForce)
+  long forceUpdateTuning = accumulateTime(_timers.forceUpdateTuning.getTotalTime());
+  long forceUpdateNonTuning = accumulateTime(_timers.forceUpdateNonTuning.getTotalTime());
+  long velocityUpdate = accumulateTime(_timers.velocityUpdate.getTotalTime());
+  long simulate = accumulateTime(_timers.simulate.getTotalTime());
+  long vtk = accumulateTime(_timers.vtk.getTotalTime());
+  long initialization = accumulateTime(_timers.initialization.getTotalTime());
+  long total = accumulateTime(_timers.total.getTotalTime());
+  long thermostat = accumulateTime(_timers.thermostat.getTotalTime());
+  long haloParticleExchange = accumulateTime(_timers.haloParticleExchange.getTotalTime());
+  long migratingParticleExchange = accumulateTime(_timers.migratingParticleExchange.getTotalTime());
+
+  if (_domainDecomposition.getDomainIndex() == 0) {
+    auto maximumNumberOfDigits = std::to_string(total).length();
+    std::cout << "Measurements:" << std::endl;
+    std::cout << timerToString("Total                        ", total, maximumNumberOfDigits, total);
+    std::cout << timerToString("  Initialization             ", initialization, maximumNumberOfDigits, simulate);
+    std::cout << timerToString("  Simulate                   ", simulate, maximumNumberOfDigits, total);
+    std::cout << timerToString("  PositionUpdate             ", positionUpdate, maximumNumberOfDigits, simulate);
+    std::cout << timerToString("  Boundaries:                ", haloParticleExchange + migratingParticleExchange,
+                               maximumNumberOfDigits, simulate);
+    std::cout << timerToString("    HaloParticleExchange     ", haloParticleExchange, maximumNumberOfDigits, simulate);
+    std::cout << timerToString("    MigratingParticleExchange", migratingParticleExchange, maximumNumberOfDigits,
+                               simulate);
+    std::cout << timerToString("  ForceUpdateTotal           ", forceUpdateTotal, maximumNumberOfDigits, simulate);
+    std::cout << timerToString("    ForceUpdatePairwise      ", forceUpdatePairwise, maximumNumberOfDigits,
+                               forceUpdateTotal);
+    std::cout << timerToString("    ForceUdpateGlobalForces  ", forceUpdateGlobalForces, maximumNumberOfDigits,
+                               forceUpdateTotal);
+    std::cout << timerToString("    ForceUpdateTuning        ", forceUpdateTuning, maximumNumberOfDigits,
+                               forceUpdateTotal);
+    std::cout << timerToString("    ForceUpdateNonTuninng    ", forceUpdateNonTuning, maximumNumberOfDigits,
+                               forceUpdateTotal);
+    std::cout << timerToString("  VelocityUpdate             ", velocityUpdate, maximumNumberOfDigits, simulate);
+    std::cout << timerToString("  Thermostat                 ", thermostat, maximumNumberOfDigits, simulate);
+    std::cout << timerToString("  Vtk                        ", vtk, maximumNumberOfDigits, simulate);
```

Comment:
```suggestion
    std::cout << timerToString("    PositionUpdate             ", positionUpdate, maximumNumberOfDigits, simulate);
    std::cout << timerToString("    Boundaries:                ", haloParticleExchange + migratingParticleExchange,
                               maximumNumberOfDigits, simulate);
    std::cout << timerToString("      HaloParticleExchange     ", haloParticleExchange, maximumNumberOfDigits, haloParticleExchange + migratingParticleExchange);
    std::cout << timerToString("      MigratingParticleExchange", migratingParticleExchange, maximumNumberOfDigits,
                               haloParticleExchange + migratingParticleExchange);
    std::cout << timerToString("    ForceUpdateTotal           ", forceUpdateTotal, maximumNumberOfDigits, simulate);
    std::cout << timerToString("      ForceUpdatePairwise      ", forceUpdatePairwise, maximumNumberOfDigits,
                               forceUpdateTotal);
    std::cout << timerToString("      ForceUdpateGlobalForces  ", forceUpdateGlobalForces, maximumNumberOfDigits,
                               forceUpdateTotal);
    std::cout << timerToString("      ForceUpdateTuning        ", forceUpdateTuning, maximumNumberOfDigits,
                               forceUpdateTotal);
    std::cout << timerToString("      ForceUpdateNonTuninng    ", forceUpdateNonTuning, maximumNumberOfDigits,
                               forceUpdateTotal);
    std::cout << timerToString("    VelocityUpdate             ", velocityUpdate, maximumNumberOfDigits, simulate);
    std::cout << timerToString("    Thermostat                 ", thermostat, maximumNumberOfDigits, simulate);
    std::cout << timerToString("    Vtk                        ", vtk, maximumNumberOfDigits, simulate);
```
- All these things are part of the simulation. Hence they need to be indented under simulation. This might mess up some spacings so check the output after applying this.
- Things under Boundaries should be analyzed relative to Boundaries and not to Simulate hence `haloParticleExchange + migratingParticleExchange`. You should probably introduce a variable for this sum.